### ISSUE TITLE
Wrap models in modules

### ISSUE THREAD

**Issue by schillic** [JuliaReach/NeuralNetworkAnalysis.jl#179]
Working with several models in the same session is not possible if they define the same constants. We should just wrap the core model parts in a function.

**Comment by schillic** [comment]
Unfortunately this is not always possible because `@taylorize` requires top-level functions.

**Comment by lbenet** [comment]
Note that `@taylorize` will not do anything with functions that it does not recognize, that is, it will pass them as they are without any possible optimization. You should give it a try.

**Comment by schillic** [comment]
But we want it to do its magic, right? Also, how does it recognize functions?

**Comment by schillic** [comment]
Here is why redefining `const`s is bad (taken from [here](https://discourse.julialang.org/t/const-variables/42956/7)):

```julia
julia> const a = 3
3

julia> f(x) = a * x
f (generic function with 1 method)

julia> f(10)
30

julia> a = 4
WARNING: redefinition of constant a. This may fail, cause incorrect answers, or produce other errors.
4

julia> f(10)
30
```

I think for our application it is fine because we redefine the functions *after* redefining the constants. But redefining constants still prints this warning, which makes the results seem untrustworthy at least.

**Comment by mforets** [comment]
In my workflow this is not an issue because I use different notebooks for different models.

**Comment by mforets** [comment]
Or the warning is something printed in the generated Literate notebooks?

**Comment by schillic** [comment]
True, but in a benchmark script we would probably run them all after each other.

**Comment by mforets** [comment]
In ReachabilityModels.jl we wrapped each Literate file into its own module as in  https://github.com/JuliaReach/ReachabilityModels.jl/blob/master/src/models/beam/beam.jl  Since it uses the `#jl` flag, it is only visible in the generated Julia script.

**Comment by schillic** [comment]
Works like a charm. I will go for that!
